tests/e2e/lib/virt_helpers.go

```diff
@@ -0,0 +1,408 @@
+	if err != nil {
+		log.Printf("Error getting HCO health: %v", err)
+		return false
+	}
+	if !ok {
+		log.Printf("HCO health field not populated yet")
+		return false
+	}
+	log.Printf("HCO health status is: %s", health)
+
+	return health == "healthy"
+}
+
+// Creates the target virtualization namespace, likely openshift-cnv or kubevirt-hyperconverged
+func (v *VirtOperator) installNamespace() error {
+	err := v.Client.Create(context.Background(), &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: v.Namespace}})
+	if err != nil {
+		log.Printf("Failed to create namespace %s: %v", v.Namespace, err)
+		return err
+	}
+	return nil
+}
+
+// Creates the virtualization operator group
+func (v *VirtOperator) installOperatorGroup() error {
+	group := &operatorsv1.OperatorGroup{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "kubevirt-hyperconverged-group",
+			Namespace: v.Namespace,
+		},
+		Spec: operatorsv1.OperatorGroupSpec{
+			TargetNamespaces: []string{
+				v.Namespace,
+			},
+		},
+	}
+	err := v.Client.Create(context.Background(), group)
+	if err != nil {
+		if !strings.Contains(err.Error(), "already exists") {
+			log.Printf("Failed to create operator group: %v", err)
+			return err
+		}
+	}
+	return nil
+}
+
+// Creates the subscription, which triggers creation of the ClusterServiceVersion.
+func (v *VirtOperator) installSubscription() error {
+	subscription := &operatorsv1alpha1.Subscription{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "hco-operatorhub",
+			Namespace: v.Namespace,
+		},
+		Spec: &operatorsv1alpha1.SubscriptionSpec{
+			CatalogSource:          "redhat-operators",
+			CatalogSourceNamespace: "openshift-marketplace",
+			Package:                "kubevirt-hyperconverged",
+			Channel:                "stable",
+			StartingCSV:            v.Csv,
+			InstallPlanApproval:    operatorsv1alpha1.ApprovalAutomatic,
+		},
+	}
+	err := v.Client.Create(context.Background(), subscription)
+	if err != nil {
+		log.Printf("Failed to create subscription: %v", err)
+		return err
+	}
+
+	return nil
+}
+
+// Creates a HyperConverged Operator instance. Another dynamic client to avoid
+// bringing in the KubeVirt APIs for now.
+func (v *VirtOperator) installHco() error {
+	resourceId := schema.GroupVersionResource{
+		Group:    "hco.kubevirt.io",
+		Resource: "hyperconvergeds",
+		Version:  "v1beta1",
+	}
+
+	unstructuredHco := unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"apiVersion": "hco.kubevirt.io/v1beta1",
+			"kind":       "HyperConverged",
+			"metadata": map[string]interface{}{
+				"name":      "kubevirt-hyperconverged",
+				"namespace": v.Namespace,
+			},
+			"spec": map[string]interface{}{},
+		},
+	}
+
+	_, err := v.Dynamic.Resource(resourceId).Namespace(v.Namespace).Create(context.Background(), &unstructuredHco, v1.CreateOptions{})
+	if err != nil {
+		log.Printf("Error creating HCO: %v", err)
+		return err
+	}
+
+	return nil
+}
+
+// Helper function to poll status with a timeout
```

Comment:
I don't understand what it waits for? When reading the code I got it, just should be more straight forward from the desc.

Reply:
I added more words, and moved it to common_utils.

Reply:
I think this function can be deleted, ginkgo has a built in function that does a very similar job

example on how to use it https://github.com/openshift/oadp-operator/blob/master/tests/e2e/lib/apps.go#L580

Reply:
I didn't see ginkgo/gomega imported in any of the other *_helpers files, so I thought maybe it wasn't supposed to be used there for some reason. I did find (via nodeagent_helpers.go) a Kubernetes API machinery module called wait, with a PollImmediate function that does pretty much the same thing. I would rather use this one than the one from gomega, but either option is definitely better than writing a new one.

Reply:
Okay, replaced this with PollImmediate from apimachinery.